server/worker/software_worker.go

```diff
@@ -335,3 +346,46 @@ func QueueBulkSetAndroidAppsAvailableForHost(
 	level.Debug(logger).Log("job_id", job.ID, "job_name", softwareWorkerJobName, "task", args.Task)
 	return nil
 }
+
+func (v *SoftwareWorker) bulkSetAndroidAppsAvailableForHosts(ctx context.Context, uuidsToIDs map[string]uint, enterpriseName string) error {
+	// for each host
+	// get the set of self-service apps that are in scope for it
+	for uuid, hostID := range uuidsToIDs {
+		appIDs, err := v.Datastore.GetAndroidAppsInScopeForHost(ctx, hostID)
+		if err != nil {
+			return ctxerr.WrapWithData(ctx, err, "get android apps in scope for host", map[string]any{"host_id": hostID})
+		}
+
+		err = v.AndroidModule.SetAppsForAndroidPolicy(ctx, enterpriseName, appIDs, map[string]string{uuid: uuid}, android.APP_STATUS_AVAILABLE)
```

Comment:
Note to whoever will do the rest of the Android app configuration story, this new worker task will need to be updated similarly to how `bulkMakeAndroidAppsAvailableForHost` is implemented on `main` to take app configs into consideration.